### ISSUE TITLE
Constraint Triggers

### ISSUE THREAD

**Issue by aidan-moore-bct** [olirice/alembic_utils#30]
Hi,

Thank you for this great library. We are currently using alembic_utils for versioning our functions and triggers.

I have been trying to use `PGTrigger` to create/update constraint triggers but from looking at the docs and code I don't believe it supports them. 

Is there any plan to extend the class to support them?

Thanks again

**Comment by olirice** [comment]
Given that `CONSTRAINT` is an optional keyword in the `CREATE [CONSTRAINT] TRIGGER`, this is a bug

I did not test this, but I'm 90% sure this work around should work for you until the bug is resolved

```python
from alembic_utils.pg_trigger import PGTrigger

class PGConstraintTrigger(PGTrigger):
    _template = "create{:s}constraint{:s}trigger{:s}{signature}{:s}{event}{:s}ON{:s}{on_entity}{:s}{action}" 
```

**Comment by aidan-moore-bct** [comment]
Thank you for the prompt response.

I had already tried inheriting `PGTrigger` and replacing the template but encountered issues with the `to_sql_statement_create` method and `from_sql`/`from_database` classmethods.

In `to_sql_statement_create` the `CREATE TRIGGER` is hard-coded:
https://github.com/olirice/alembic_utils/blob/1d08185002e854a66aba9e928c71077ff76f381e/src/alembic_utils/pg_trigger.py#L86
So including the optional `CONSTRAINT` keyword still results in an ordinary trigger. An auto-generated migration is successfully created but when attempting to update the database I encounter the following exception:
```console
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "DEFERRABLE"
LINE 1: ...ot_version_AI AFTER INSERT ON public.table_name DEFERRABLE...
                                                             ^

[SQL: CREATE TRIGGER trigger_name AFTER INSERT ON public.table_name DEFERRABLE INITIALLY DEFERRED
    FOR EACH ROW
    EXECUTE PROCEDURE func_name ();]
```
This exception occurs because `DEFERRABLE INITIALLY DEFERRED` requires a constraint trigger. As a workaround I overrode `to_sql_statement_create` with:
```python
def to_sql_statement_create(self) -> str:
    statement = super().to_sql_statement_create()
    statement.text = statement.text.replace("CREATE TRIGGER", "CREATE CONSTRAINT TRIGGER")
    return statement
```
I could then autogenerate a new migration and successfully upgrade and downgrade.

The issue with `from_sql`/`from_database` came when attempting to create a subsequent auto-generated migration when constraint triggers had already been added to the database, which results in this exception:
```console
    raise SQLParseFailure(f'Failed to parse SQL into PGTrigger """{sql}"""')
alembic_utils.exceptions.SQLParseFailure: Failed to parse SQL into PGTrigger """CREATE CONSTRAINT TRIGGER trigger_name AFTER INSERT ON public.table_name DEFERRABLE INITIALLY DEFERRED FOR EACH ROW EXECUTE PROCEDURE func_name()"""
```
Looking at `from_database` I found:
https://github.com/olirice/alembic_utils/blob/1d08185002e854a66aba9e928c71077ff76f381e/src/alembic_utils/pg_trigger.py#L121
Because this list comprehension uses `PGTrigger` the model redefined in `PGConstraintTrigger` is ignored. 

This means that `result` in `from_sql` is `None` for a constraint trigger and results in the `SQLParseFailure` exception.
https://github.com/olirice/alembic_utils/blob/1d08185002e854a66aba9e928c71077ff76f381e/src/alembic_utils/pg_trigger.py#L34-L59
Once the template and logic in `PGTrigger` is altered to handle constraint triggers I think this second issue will resolve itself.

I hope this is useful when implementing a fix.

**Comment by olirice** [comment]
Thank you for the detail & sorry for the slow reply

`PGTrigger` now has an `is_constraint` argument for constraint triggers
In docs: https://olirice.github.io/alembic_utils/api/
In src: https://github.com/olirice/alembic_utils/blob/master/src/alembic_utils/pg_trigger.py#L38

The update is included in pypi version `alembic_utils>=0.3.13`

**Comment by aidan-moore-bct** [comment]
I have just updated our version of `alembic_utils` and have converted our code to include the `is_constraint` and `on_entity` arguments and it is working exactly as I expected.

Thank you for taking the time to fix this.
